Install fail

Hi, I tried installation both with repo and compiling but every time I have this error

`error[E0277]: the trait bound `ServerOptions: From<([{integer}; 4], u16)>` is not satisfied
   --> src/main.rs:143:16
    |
143 |     let addr = ([0, 0, 0, 0], bind).into();
    |                ^^^^^^^^^^^^^^^^^^^^ ---- required by a bound introduced by this call
    |                |
    |                the trait `From<([{integer}; 4], u16)>` is not implemented for `ServerOptions`
    |
    = note: required for `([{integer}; 4], u16)` to implement `Into<ServerOptions>`

error[E0277]: `ServerOptions` doesn't implement `std::fmt::Display`
   --> src/main.rs:145:43
    |
145 |     log::info!("starting exporter on {}", addr);
    |                                           ^^^^ `ServerOptions` cannot be formatted with the default formatter
    |
    = help: the trait `std::fmt::Display` is not implemented for `ServerOptions`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: this error originates in the macro `format_args` which comes from the expansion of the macro `log::info` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `prometheus_folder_size_exporter` due to 2 previous errors
error: failed to compile `prometheus_folder_size_exporter v0.5.0 (/home/XXXXXX/prometheus_folder_size_exporter)`, intermediate artifacts can be found at `/home/XXXXXXX/prometheus_folder_size_exporter/target`
`

On Debian with
cargo 1.67.0 (8ecd4f20a 2023-01-10)
rustc 1.67.0 (fc594f156 2023-01-24)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name           = "prometheus_folder_size_exporter"
-version        = "0.5.0"
+version        = "0.5.1"
 authors        = ["Francesco Cogno <francesco.cogno@outlook.com>"]
 description    = "Prometheus Folder Size Exporter"
 edition        = "2018"
@@ -27,4 +27,3 @@ hyper                    = {version = "0.14", features = ["server"] }
 http                     = "0.2.1"
 tokio                    = { version = "1.4", features = ["full"] }
 prometheus_exporter_base = { version = "1.2.0", features = ["hyper_server"] }
-

--- a/src/exporter_error.rs
+++ b/src/exporter_error.rs
@@ -12,7 +12,7 @@ pub(crate) enum ExporterError {
     UTF8 { e: std::string::FromUtf8Error },
 
     #[error("JSON format error: {}", e)]
-    JSON { e: serde_json::error::Error },
+    Json { e: serde_json::error::Error },
 
     #[error("IO Error: {}", e)]
     IO { e: std::io::Error },
@@ -44,6 +44,6 @@ impl From<std::string::FromUtf8Error> for ExporterError {
 
 impl From<serde_json::error::Error> for ExporterError {
     fn from(e: serde_json::error::Error) -> Self {
-        ExporterError::JSON { e }
+        ExporterError::Json { e }
     }
 }

--- a/src/folder_scanner.rs
+++ b/src/folder_scanner.rs
@@ -120,8 +120,8 @@ impl FolderScanner {
             log::trace!("scanning folder {:?}", folder);
 
             match folder.explode_depth {
-                0 => v_sizes.push(scan_folder_sum(&folder)?),
-                _ => v_sizes.extend_from_slice(&scan_folder_explode(&folder)?),
+                0 => v_sizes.push(scan_folder_sum(folder)?),
+                _ => v_sizes.extend_from_slice(&scan_folder_explode(folder)?),
             }
         }
         Ok(v_sizes)

--- a/src/folder_scanner.rs
+++ b/src/folder_scanner.rs
@@ -120,8 +120,20 @@ impl FolderScanner {
             log::trace!("scanning folder {:?}", folder);
 
             match folder.explode_depth {
-                0 => v_sizes.push(scan_folder_sum(folder)?),
-                _ => v_sizes.extend_from_slice(&scan_folder_explode(folder)?),
+                0 => scan_folder_sum(folder)
+                    .map(|res| v_sizes.push(res))
+                    .unwrap_or_else(|err| {
+                        log::warn!("skipping summing {:?} due to error {:?}", folder.path, err)
+                    }),
+                _ => scan_folder_explode(folder)
+                    .map(|res| v_sizes.extend_from_slice(&res))
+                    .unwrap_or_else(|err| {
+                        log::warn!(
+                            "skipping exploding {:?} due to error {:?}",
+                            folder.path,
+                            err
+                        )
+                    }),
             }
         }
         Ok(v_sizes)

--- a/src/folder_scanner.rs
+++ b/src/folder_scanner.rs
@@ -47,7 +47,15 @@ fn scan_folder_explode(
             if folder_to_scan.explode_depth == -1 || folder_to_scan.explode_depth > 1 {
                 v.extend_from_slice(&scan_folder_explode(&folder_inner)?);
             } else {
-                v.push(scan_folder_sum(&folder_inner)?);
+                scan_folder_sum(&folder_inner)
+                    .map(|res| v.push(res))
+                    .unwrap_or_else(|err| {
+                        log::warn!(
+                            "scan_folder_sum({}) failed due to error {:?}",
+                            folder_inner.path,
+                            err
+                        )
+                    })
             }
         } else {
             tot += entry.metadata()?.len();
@@ -67,7 +75,13 @@ fn scan_folder_sum(folder_to_scan: &FolderToScan) -> Result<FolderWithSize, std:
     log::trace!("scan_folder_sum({:?})", &folder_to_scan);
     let mut tot: u64 = 0;
 
+    log::trace!("before read_dir({:?})", folder_to_scan.path);
     for entry in read_dir(&folder_to_scan.path)? {
+        log::trace!(
+            "after read_dir({:?}) == entry {:?}",
+            folder_to_scan.path,
+            entry
+        );
         let entry = entry?;
 
         let folder_inner = FolderToScan {
@@ -78,9 +92,11 @@ fn scan_folder_sum(folder_to_scan: &FolderToScan) -> Result<FolderWithSize, std:
         };
 
         if entry.file_type()?.is_dir() && folder_to_scan.sum_remaining_subfolders {
-            tot += scan_folder_sum(&folder_inner)?.size;
+            tot += scan_folder_sum(&folder_inner)
+                .map(|res| res.size)
+                .unwrap_or(0);
         } else {
-            tot += entry.metadata()?.len();
+            tot += entry.metadata().map(|res| res.len()).unwrap_or(0);
         }
     }
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -18,7 +18,7 @@ use std::time::Duration;
 
 async fn perform_request(
     _req: http::request::Request<Body>,
-    state: Arc<Arc<State>>,
+    state: Arc<State>,
 ) -> Result<String, Box<dyn Error + Send + Sync>> {
     let results = if state.options.background_poll_seconds.is_some() {
         // loop until we have some data.
@@ -122,7 +122,7 @@ async fn main() {
     env_logger::init();
     log::info!("using options: {:?}", options);
 
-    let state = Arc::new(State::new(options));
+    let state = State::new(options);
 
     // start the background thread only
     // if requested
@@ -139,12 +139,16 @@ async fn main() {
     }
 
     let bind = matches.value_of("port").unwrap();
-    let bind = u16::from_str_radix(&bind, 10).expect("port must be a valid number");
+    let bind = bind.parse::<u16>().expect("port must be a valid number");
     let addr = ([0, 0, 0, 0], bind).into();
-
     log::info!("starting exporter on {}", addr);
 
-    prometheus_exporter_base::render_prometheus(addr, state, |request, state| {
+    let server_options = ServerOptions {
+        addr,
+        authorization: Authorization::None,
+    };
+
+    prometheus_exporter_base::render_prometheus(server_options, state, |request, state| {
         Box::pin(perform_request(request, state))
     })
     .await

--- a/src/options.rs
+++ b/src/options.rs
@@ -4,7 +4,6 @@ use std::io::Read;
 
 #[derive(Debug, Clone)]
 pub(crate) struct Options {
-    pub folders_file: String,
     pub verbose: bool,
     pub folders_to_scan: FolderScanner,
     pub background_poll_seconds: Option<std::time::Duration>,
@@ -17,13 +16,12 @@ impl Options {
             .value_of("background_poll_seconds")
             .map(|s| std::time::Duration::from_secs(s.parse().unwrap()));
 
-        let mut file = File::open(&folders_file).unwrap();
+        let mut file = File::open(folders_file).unwrap();
         let mut file_contents = String::new();
 
         file.read_to_string(&mut file_contents).unwrap();
 
         Options {
-            folders_file,
             verbose: matches.is_present("verbose"),
             folders_to_scan: FolderScanner::from_json(&file_contents).unwrap(),
             background_poll_seconds,

